src/evaluators/ForStatement.js

```diff
@@ -192,6 +196,7 @@ export default function(
         bodyResult = ForBodyEvaluation(realm, test, update, body, perIterationLets, labelSet, strictCode);
       } finally {
         // 12. Set the running execution context's LexicalEnvironment to oldEnv.
+        realm.onDestroyScope(realm.getRunningContext().lexicalEnvironment);
```

Comment:
Might this be different from loopEnv? If so, how did that come about and could those circumstances lead to environments that become inaccessible because of the next line, but that are not destroyed? If not, it would be good to assert that here using an invariant.